config/config.go

```diff
@@ -303,7 +355,7 @@ func (s *HeaderMatch) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return errors.New("header name must be set for HTTP header matchers")
 	}
 
-	if s.Regexp == "" {
+	if s.Regexp.Regexp == nil {
```

Comment:
Isn't this never nil, and you should be checking for the empty string?

I think you're being inconsistent in how you're treating this.

Reply:
It's possible that `s.Regexp.Regexp` is nil when following case. (http.fail_if_header_not_matches.regexp isn't explicitly specified.)
```
modules:
  http_headers:
    prober: http
    timeout: 5s
    http:
      fail_if_header_not_matches:
        - header: Access-Control-Allow-Origin
          allow_missing: false
```
And as you said I have to check for the empty string, so I added its condition.